instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/internal/JettyHttpClient9TracingInterceptor.java

```diff
@@ -135,14 +131,11 @@ private void startSpan(Request request) {
   @Override
   public void onBegin(Request request) {}
 
-  @Override
-  public void onComplete(Result result) {
-    closeIfPossible(result.getResponse());
-  }
-
   @Override
   public void onSuccess(Response response) {
-    closeIfPossible(response);
+    if (this.context != null) {
+      instrumenter.end(this.context, response.getRequest(), response, null);
+    }
```

Comment:
```suggestion
    if (this.context != null) {
      instrumenter.end(this.context, response.getRequest(), response, null);
    } else {
      logger.fine("onSuccess - could not find an otel context");
    }
```

Reply:
Is this log line really useful? Looking at the code currently context being null doesn't really indicate that anything is wrong. Imo it would be better to refactor this so that the listener isn't added at all when span isn't started.

Reply:
> Imo it would be better to refactor this so that the listener isn't added at all when span isn't started.
👍